chatwoot/chatwoot
# fix: bots included in time to response metrics

## Description

Time to first response included the metrics from bots as well, this PR fixes that. 

The changes in this PR

1. ~https://github.com/chatwoot/chatwoot/pull/6409/commits/bb9171945d5e3b2f6015f4f96dd1b76b3efb6987 & https://github.com/chatwoot/chatwoot/pull/6409/commits/de1977c219a2e7a9180dd02272244fe3b3f7ce89 ignore bot values in conversations, **this will affect the current values the users see on their reports**, we can probably remove this, need thoughts?~ Reverted these commits
1. https://github.com/chatwoot/chatwoot/pull/6409/commits/4a236930f8f9cc036d9e6e9f2a956b3740455193 & https://github.com/chatwoot/chatwoot/pull/6409/commits/9f4cfea4d787b62f889ea03325cf85866170e9ec logs the bot handoff event explicitly in the `response_events` table
1. https://github.com/chatwoot/chatwoot/pull/6409/commits/33f434ac72288b1a68a73bdd71a188d3caa1ec93 ensure that `FIRST_REPLY_CREATED` is dispatched only if the response is made by a human and not `AgentBot`, **this makes the first change redundant over time as new data accumulates**
1. https://github.com/chatwoot/chatwoot/pull/6409/commits/606d0b0f469611f7d03d6004766800bd4816273a this ensures that the bot_handoff time is taken into account when calculating the first_response. 

Issue for reference: https://github.com/chatwoot/product/issues/790

## Type of change


## How Has This Been Tested?

- Setup CSML locally, and add a basic bot.
- Try starting a conversation from the widget using [the test page](http://localhost:3000/widget_tests)
- Observe the time to response time in the conversation report

Apply this patch and try it again, it should disregard the bot conversations.

## Checklist:
